src/convertToObject.js

```diff
@@ -1,18 +1,22 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';')
+    .filter(style => style.trim().length > 0)
+    .reduce((styles, style) => {
+      const [nameOfStyle, valueOfStyle] = style
+        .split(':')
+        .map(stringPart => stringPart.trim());
+
+      styles[nameOfStyle] = valueOfStyle;
+
+      return styles;
```

Comment:
```suggestion
      styles[nameOfStyle] = valueOfStyle;

      return {
        ...styles,
        [nameOfStyle]: valueOfStyle
      };
```

Reply:
Not critical in this case, but still it's a bad idea to mutate incoming function param